packages/features/bookings/lib/getBookingResponsesSchema.ts

```diff
@@ -276,8 +276,10 @@ function preprocess<T extends z.ZodType>({
         }
 
         if (bookingField.type === "phone") {
-          if (!bookingField.hidden && !(await phoneSchema.safeParseAsync(value)).success) {
-            ctx.addIssue({ code: z.ZodIssueCode.custom, message: m("invalid_number") });
+          if (!bookingField.hidden && (isRequired || (value && value.trim() !== ""))) {
```

Comment:
it's difficult to understand what this conditional does here. I suggest:
1. extract `(isRequired || (value && value.trim() !== "")` into a new variable
2. leave explanatory comments